stratum/hal/lib/common/p4_service.cc

```diff
@@ -57,13 +58,21 @@ namespace hal {
 P4Service::P4Service(OperationMode mode, SwitchInterface* switch_interface,
                      AuthPolicyChecker* auth_policy_checker,
                      ErrorBuffer* error_buffer)
+    : P4Service(mode, switch_interface, auth_policy_checker, error_buffer,
+                TargetOptions::default_target_options) {}
+
+P4Service::P4Service(OperationMode mode, SwitchInterface* switch_interface,
```

Comment:
Why not have just one constructor like below?
P4Service::P4Service(OperationMode mode, SwitchInterface* switch_interface,
                     AuthPolicyChecker* auth_policy_checker,
                     ErrorBuffer* error_buffer,
                     const TargetOptions& target_options =  TargetOptions::default_target_options)